SharedProcessors/MunkiAutoStaging.py

```diff
@@ -0,0 +1,298 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""See docstring for MunkiAutoStaging class"""
+
+from __future__ import absolute_import
+
+import os
+import plistlib
+import glob
+
+from datetime import datetime
+from distutils.version import StrictVersion
+
+from autopkglib import Processor, ProcessorError
+from autopkglib.munkirepolibs.AutoPkgLib import AutoPkgLib
+from autopkglib.munkirepolibs.MunkiLib import MunkiLib
+
+__all__ = ["MunkiAutoStaging"]
+
+
+def _find_matching_item(repo_library, name):
+    """Looks through all catalog for items matching the named one.
+    Returns a list of all matching items if found."""
+
+    pkgdb = repo_library.make_catalog_db()
+
+    if not name:
+        return pkgdb["items"]
+
+    matching_items = []
+
+    for item in pkgdb["items"]:
+        if item["name"] == name:
+            matching_items.append(item)
+
+    return matching_items
+
+
+def _fetch_repo_library(
+    munki_repo,
+    munki_repo_plugin,
+    munkilib_dir,
+    repo_subdirectory,
+    force_munki_lib,
+):
+
+    if munki_repo_plugin == "FileRepo" and not force_munki_lib:
+        return AutoPkgLib(munki_repo, repo_subdirectory)
+
+    return MunkiLib(
+        munki_repo, munki_repo_plugin, munkilib_dir, repo_subdirectory
+    )
+
+
+class MunkiAutoStaging(Processor):
+    """This processor will automatically move all given Munki items from a testing
+    catalog to a production catalog after a given amount of days."""
+
+    description = __doc__
+    input_variables = {
+        "MUNKI_REPO": {
+            "description": "Path to a mounted Munki repo.",
+            "required": True,
+        },
+        "MUNKI_REPO_PLUGIN": {
+            "description": (
+                "Munki repo plugin. Defaults to FileRepo. Munki must be "
+                "installed and available at MUNKILIB_DIR if a plugin other "
+                "than FileRepo is specified."
+            ),
+            "required": False,
+            "default": "FileRepo",
+        },
+        "MUNKI_REPO_SUBDIR": {
+            "description": (
+                "Subdirectory of Munki repo, useful for large repositories."
+            ),
+            "required": False,
+            "default": "",
+        },
+        "MUNKILIB_DIR": {
+            "description": (
+                "Directory path that contains munkilib. Defaults to "
+                "/usr/local/munki"
+            ),
+            "required": False,
+            "default": "/usr/local/munki",
+        },
+        "MUNKI_PKGINFO_FILE_EXTENSION": {
+            "description": (
+                "Extension for output pkginfo files. Default is 'plist'.",
+            ),
+            "required": False,
+            "default": "plist",
+        },
+        "MUNKI_TESTING_CATALOG": {
```

Comment:
This variable should be called MUNKI_STAGING_CATALOG to make the connection to variable line 124 MUNKI_STAGING_DAYS more obvious.